docs/overview.adoc

```diff
@@ -0,0 +1,331 @@
+      "text": "…"
+    },
+    "relationships": {
+      "comments": {
+        "data": [
+          {"id": 42, "type": "comment"}
+        ]
+      }
+    }
+  },
+  "included": [
+    {
+      "id": 42,
+      "type": "comment",
+      "attributes": {
+        "text": "…",
+        "approved": true
+      }
+    }
+  ]
+}
+----
+
+So different requests need to access the same data.
+However, taking a relational database as example, accessing the data may not be as simple as generating a SQL `SELECT` query from a request like `GET /article/42` and applying an ORM mapping to the resoult. Instead, there are multiple additional factors to consider:
+
+* Where does the data for the resources come from, what if articles and comments are stored in different databases?
+* Do we need to limit the access to individual resources based on authorization? E.g. moderators being able to see all comments but normal users only seeing approved ones.
+* How can we avoid code duplication and even more importantly ensure that the same access limitations are applied, when accessing the same resources via different requests (e.g. a comment via `GET /article/42?include=comments` or `GET /comment/42`)?
+* How can we limit generic boilerplate code when allowing access to (many) different resource types?
+
+=== Coupling resource types to logic and behavior
+
+This library addresses the problem by applying an object-oriented approach.
+A "resource type" is not considered just a string to uniquely identify corresponding resources, but an expert in retrieving and manipulating corresponding resources.
+The idea is that regardless of the way a resource (e.g. a comment) is retrieved, ultimately it is always retrieved via a resource type instance that is tailored in retrieving resources of that type.
+This allows to abstract away all details (like access limitations and access to the data source) within the resource type instance.
+
+To make this concept more tangible, we use the process of fetching a single article via `GET /article/42` as example.
+Accessing a single resource via its identifier like that requires a corresponding `GetableTypeInterface` instance to be present.
+Classes inheriting from `GetableTypeInterface` must implement three methods:
+
+* `getTypeName(): non-empty-string`
+* `getEntity(non-empty-string $identifier): object`
+* `getTransformer(): TransformerAbstract`
+
+For this example use-case (`GET /article/42`), the high-level process is relatively simple:
+
+1. The correct `GetableTypeInterface` instances is retrieved, corresponding to `comment` resources. How this is done is left to the application. For example within Symfony, each resource type string may be given its own route and within that route the retrieval of the correct `GetableTypeInterface` instance is hardcoded. Or a single route for all resource types may exist, dynamically retrieving the correct instance via the resource type string in the URL (i.e. `article`) from a dictionary.
```

Comment:
I thought this example is about `article`, why is `comment` the corresponding resource?

Reply:
Should have been `article`. Fixed.